### PR TITLE
Add websocket keepalive

### PR DESCRIPTION
**Pull request by kustov-an**
My attempt to implement the ping-pong keepalive mechanism. Since I didn't want to break the compatibility, the timeout is set to constant 60 seconds.

### CODE REVIEW

**Comment by kustov-an** [comment]
#29

File: websocket.go [hunk 1]
```diff
@@ -51,3 +55,28 @@ var wsServe = func(cfg *wsConfig, handler WsHandler, errHandler ErrHandler) (don
...
+	lastResponse := time.Now()
+	c.SetPongHandler(func(msg string) error {
+		lastResponse = time.Now()
+		return nil
+	})
+
+	go func() {
+		defer ticker.Stop()
+		for {
+			err := c.WriteMessage(websocket.PingMessage, []byte("keepalive"))
```
Comment (adshao) [inline websocket.go]: use c.WriteControl to send a ping message without data: https://godoc.org/github.com/gorilla/websocket#Conn.WriteControl

File: websocket.go [hunk 2]
```diff
@@ -35,6 +38,7 @@ var wsServe = func(cfg *wsConfig, handler WsHandler, errHandler ErrHandler) (don
 			}
 		}()
 		defer close(doneC)
+		keepAlive(c, cfg.timeout)
```
Comment (adshao) [inline websocket.go]: it's better to enable keepAlive as an config option, may be we should disable it by default

File: websocket_service.go [hunk 3]
```diff
@@ -24,7 +26,7 @@ type WsPartialDepthHandler func(event *WsPartialDepthEvent)
 // WsPartialDepthServe serve websocket partial depth handler with a symbol
 func WsPartialDepthServe(symbol string, levels string, handler WsPartialDepthHandler, errHandler ErrHandler) (doneC, stopC chan struct{}, err error) {
 	endpoint := fmt.Sprintf("%s/%s@depth%s", baseURL, strings.ToLower(symbol), levels)
-	cfg := newWsConfig(endpoint)
+	cfg := newWsConfig(endpoint, timeout)
```
Comment (adshao) [inline websocket_service.go]: if timeout cannot be changed, it's not a good idea to pass it as a required argument to `newWsConfig`, use optional argument instead?

**Comment by kustov-an** [comment]
Made keepalive optional. I didn't want to break the api, so I added two package-level variables.

**Pull request merged**